src/alpaka/plugin-SiPixelClusterizer/alpaka/SiPixelFedCablingMapESProducer.cc

```diff
@@ -33,15 +33,15 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
 
     auto cablingMap_h{cms::alpakatools::createHostView<SiPixelFedCablingMapGPU>(&obj, 1u)};
     auto cablingMap_d{cms::alpakatools::allocDeviceBuf<SiPixelFedCablingMapGPU>(1u)};
+    alpaka::prepareForAsyncCopy(cablingMap_d);
     alpaka::memcpy(queue, cablingMap_d, cablingMap_h, 1u);
     eventSetup.put(std::make_unique<SiPixelFedCablingMapGPUWrapper>(std::move(cablingMap_d), true));
 
     auto modToUnp_h{cms::alpakatools::createHostView<unsigned char>(modToUnpDefault.data(), modToUnpDefSize)};
     auto modToUnp_d{cms::alpakatools::allocDeviceBuf<unsigned char>(modToUnpDefSize)};
+    alpaka::prepareForAsyncCopy(modToUnp_d);
     alpaka::memcpy(queue, modToUnp_d, modToUnp_h, modToUnpDefSize);
 
-    alpaka::wait(queue);
```

Comment:
this should not be removed !

Reply:
Ha yes true, it is not precisely that the `queue` is destructed, but just that the `queue` destructor _returns_.
https://github.com/alpaka-group/alpaka/blob/0.7.0/include/alpaka/queue/cuda_hip/QueueUniformCudaHipRtBase.hpp#L80
With no `wait`, the kernels start with no guarantee that the asynchronous mem copies are completed properly.

>Maybe we can suggest to the Alpaka developers that a Queue should wait for the work queued onto it on destruction ?
Maybe, but maybe in a sense we do not want a _local_ queue to be destructed, and wait _here_ anyway? 
I was thinking all these local queues (streams) should be removed once the `ScopedContext` is implemented: here the only thing we want is simply to have the data copies and the kernels that need the data on the _same_ stream?

Like in the ScopedContext PR, also maybe adding the equivalent of `SiPixelFedCablingMapGPUWrapper::getGPUProductAsync` (https://github.com/cms-patatrack/pixeltrack-standalone/blob/master/src/cuda/CondFormats/SiPixelFedCablingMapGPUWrapper.cc#L27 ), and using it here: https://github.com/cms-patatrack/pixeltrack-standalone/pull/236/files#diff-04a70eb4f8dece3a473ca1100b50cd03da79257a1d6434d83607de8c106b23d2R73 
?